core/src/main/scala/quasar/plugin/jdbc/datasource/JdbcDatasourceModule.scala

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2020 Precog Data
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.plugin.jdbc.datasource
+
+import quasar.plugin.jdbc.{ManagedTransactor, Redacted, TransactorConfig}
+
+import java.lang.{Exception, String}
+
+import scala.StringContext
+import scala.concurrent.ExecutionContext
+import scala.util.{Either, Left, Random, Right}
+
+import argonaut._, Argonaut._, ArgonautCats._
+
+import cats.Hash
+import cats.data.{EitherT, NonEmptyList}
+import cats.effect._
+import cats.implicits._
+
+import doobie._
+
+import quasar.RateLimiting
+import quasar.api.datasource.{DatasourceError => DE}
+import quasar.connector.{ByteStore, MonadResourceErr}
+import quasar.connector.datasource.LightweightDatasourceModule
+
+import org.slf4s.{Logger, LoggerFactory}
+
+/** A Quasar LightweightDatsourceModule for JDBC sources.
+  *
+  * Handles boilerplate common to all JDBC datasources, such as
+  *   - parsing JSON into a vendor-specific config
+  *   - constructing a pooled `Transactor` along with the necessary threadpools
+  *   - validating a connection to the database can be established
+  *   - logging
+  */
+abstract class JdbcDatasourceModule[C: DecodeJson] extends LightweightDatasourceModule {
```

Comment:
Will we be able to (or want to) port any of our existing datasources over to use this?

Reply:
Yes, ideally all of them that use JDBC. We'll also want to add support for destinations and convert those as well.